pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+    # vec[update_dir] will not be 0
+    if vec[update_dir] > 0:
+        idxs[update_dir] += 1
+    elif vec[update_dir] < 0:
+        idxs[update_dir] -= 1 
+    updated_point = np.add(point, dist_min * vec)
+    dist_temp += dist_min
+    return idxs, updated_point, dist_temp
+
+def _is_ray_on_boundary(start, vec, bounds):
+    """
+    Check whether the ray on a boundary surface.
+
+    Parameters:
+    -----------
+    start : numpy array
+        Start point coordinate.
+    vec : numpy array
+        Direction vector.
+    bounds : numpy array
+        Boundaries of the mesh grid.
+
+    Return:
+    -------
+    True : bool
+        Ray is on a boundary surface
+    False : bool
+        Ray is not on a boundary surface
+    """
+    if 1.0 in vec:
+        for idx in np.argwhere(vec!=1.0).flatten():
+            if start[idx] in bounds[idx]:
+                return True
+    return False
+
+def _ray_voxel_traverse(bounds, start, end):
+    """
+    Ray (line segment) traversal in structured mesh.
+    Return the voxels that intersect with the line segment.
+
+    Parameters:
+    -----------
+    bounds : numpy array
+        Boundaries of the mesh grid.
+    start : numpy array
+        Start point.
+    end : numpy array
+        End point.
+
+    Return:
+    -------
+    voxles : list of tuples
+        List of voxel indices. These voxels intersect with the line segment.
+    """
+    # Initialization
+    # Find the voxel that start point in
+    point = np.copy(start)
+    idxs = [-1, -1, -1]
+    voxels = set()
+    vec = _calc_vec_dir(start, end)
+    # if the ray on a boundary, then return empty set
+    if _is_ray_on_boundary(start, vec, bounds):
+        return set()
+    dist_end = np.linalg.norm(end-start)
+    dist_temp = 0.0
+    while dist_temp < dist_end:
+        if _is_point_in_mesh(bounds, point):
+            # the point is in the mesh
+            # calculate the voxel id
+            for dr in range(len(idxs)):
+                idxs[dr] = _find_voxel_idx_1d(bounds[dr], point[dr], vec[dr])
+            if -1 in idxs:
+                # point on mesh boundary with direction toward outside the mesh
+                return voxels
+            # add current voxel
+            voxels.add((idxs[0], idxs[1], idxs[2]))
+            dist_maxs = _calc_max_travel_length_in_current_voxel(
+                point, end, vec, bounds)
+            idxs, point, dist_temp = _move_to_next_voxel(
+                idxs, point, dist_temp, dist_maxs, vec)
+        else:
+            # Check whether the ray intersecs the mesh. Calculate the maximum travel
+            # length of the ray to the outside boundaries of the mesh. If all the
+            # lengths are negtive, then there is no intersection, return. Ohterwise,
+            # choose the minimum length and move the point to that position.
+            # left, right, back, front, down, up
+            dist_maxs = [0.0] * 6
+            for dr in [0, 1, 2]:
```

Comment:
For consistency, why not
```suggestion
            for dr in range(len(idxs):
```